### PR TITLE
!str #17393: Make stream-tests pass with serialize-messages=on

### PR DESCRIPTION
**Pull request by drewhk**
Please note that ActorPublisher and ActorSubscriber messages are no longer having a SerialVersionUID and I think they should not be allowed to be serialized. Most of the Subscriptions and Subscribers are not serializable therefore the handshake won't work anyway in quite a lot of the cases.
- [x] Http

### CODE REVIEW

File: akka-http-core/src/main/scala/akka/http/impl/engine/client/PoolGateway.scala [hunk 1]
```diff
@@ -40,7 +40,7 @@ private[http] class PoolGateway(hcps: HostConnectionPoolSetup,
 
   private val state = {
     val shutdownCompletedPromise = Promise[Unit]()
-    val props = Props(new PoolInterfaceActor(hcps, shutdownCompletedPromise, this))
+    val props = Props(new PoolInterfaceActor(hcps, shutdownCompletedPromise, this)).withDeploy(Deploy.local)
```
Comment (2m) [inline akka-http-core/src/main/scala/akka/http/impl/engine/client/PoolGateway.scala]: Why are these needed? Props are serializable with `NoScopeGiven` as well.

**Comment by 2m** [comment]
LGTM

File: akka-http-core/src/main/scala/akka/http/impl/engine/client/PoolGateway.scala (hunk 1 above)
Comment (drewhk) [inline akka-http-core/src/main/scala/akka/http/impl/engine/client/PoolGateway.scala]: This explicitly states that these are deployed locally.

**Comment by drewhk** [comment]
Fixes #17393
